sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_operations_base.py

```diff
@@ -114,6 +115,26 @@ def begin_create(
 
         workspace.resource_group = resource_group
         template, param, resources_being_deployed = self._populate_arm_paramaters(workspace, **kwargs)
+        # check if create with workspace hub request is valid
+        if workspace._kind == PROJECT_WORKSPACE_KIND and workspace.workspace_hub:
+            if not all(
+                x is None
+                for x in [
+                    workspace.storage_account,
+                    workspace.container_registry,
+                    workspace.key_vault,
+                    workspace.public_network_access,
+                    workspace.managed_network,
+                    workspace.customer_managed_key,
+                ]
+            ):
+                msg = "To use workspaceHub, only specify location, application insight, encryption and identity."
```

Comment:
The error message is not clear. Please update it to:
To create a project workspace with a workspace hub, please only specify name, description, display_name, location, application insight, encryption and identity."

Reply:
I just noticed we use workspaceHub's encryption to override project workspace, I will make sure encryption is None as well.